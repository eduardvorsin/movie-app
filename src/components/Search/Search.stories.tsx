import { Meta, StoryObj } from '@storybook/react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { action } from '@storybook/addon-actions';
import Search, { Props } from './Search';
import { ChangeEventHandler, FormEventHandler, useState } from 'react';
import { useTranslation } from 'react-i18next';

const meta: Meta<typeof Search> = {
	title: 'components/Search',
	tags: ['autodocs'],
	argTypes: {
		className: {
			description: 'Additional classes for the Search'
		},
		testId: {
			description: 'Id for testing the component',
		},
		label: {
			description: 'Text description for the input field',
		},
		isDisabled: {
			description: 'Disabling the input and the button makes it non-functional',
		},
		isInvalid: {
			description: 'Visually highlights the input in red when incorrectly entered data',
		},
		isReadOnly: {
			description: 'Makes the input read-only',
		},
		name: {
			description: 'Name for the input',
		},
		placeholder: {
			description: 'The filling text that is displayed when the input is empty',
		},
		onChange: {
			description: 'The callback is executed when the input value changes',
			control: false,
		},
		onFocus: {
			description: 'The callback is executed when the focus hits the input',
			control: false,
		},
		onBlur: {
			description: 'The callback is executed when the input loses focus',
			control: false,
		},
		onSubmit: {
			description: 'The callback is executed when the search form is submitted',
			control: false,
		},
		value: {
			description: 'Input value',
		},
		labelHidden: {
			description: 'Makes the label visible only to screen readers',
		},
		error: {
			description: 'Error Message',
		},
		id: {
			description: 'Component id',
		}
	},
	component: Search,
	parameters: {
		docs: {
			description: {
				component: 'The search component with hints is an interactive element that helps users find information quickly and conveniently. As you enter a query into the search bar, the component offers automatically generated search suggestions based on previous queries or popular queries.',
			},
		},
	},
}

export default meta;
type Story = StoryObj<typeof Search>;

const SearchWithHooks = (props: Omit<Props, 'onChange' | 'value' | 'dictionary'>) => {
	const [value, setValue] = useState<string>('');
	const { t } = useTranslation('common');
	const dictionary = { button: t('search.button') };

	const changeHandler: ChangeEventHandler<HTMLInputElement> = (e) => {
		setValue(e.currentTarget.value);
		action('Changed')(e);
	}

	const submitHandler: FormEventHandler<HTMLFormElement> = (e) => {
		e.preventDefault();
		if (props.onSubmit) props.onSubmit(e);
	}

	return (
		<Search
			{...props}
			dictionary={dictionary}
			className={`max-w-[13rem] ${props.className}`}
			value={value}
			onChange={changeHandler}
			onSubmit={submitHandler}
		/>
	)
};

export const Default: Story = {
	args: {
		label: 'Default',
		name: 'Default',
		id: 'Default',
	},
	parameters: {
		docs: {
			description: {
				story: 'Default state of the search input',
			},
		},
	},
	render: (args) => (<SearchWithHooks {...args} />),
};

export const Disabled: Story = {
	args: {
		label: 'Disabled',
		name: 'Disabled',
		id: 'Disabled',
		isDisabled: true,
	},
	parameters: {
		docs: {
			description: {
				story: 'Apply this to actions that are currently unavailable. The user interface should effectively convey why the input is not active and what steps are needed to activate it.',
			},
		},
	},
	render: (args) => (<SearchWithHooks {...args} />)
};

export const Invalid: Story = {
	args: {
		label: 'Invalid',
		name: 'Invalid',
		id: 'Invalid',
		isInvalid: true,
	},
	parameters: {
		docs: {
			description: {
				story: 'It is used to visually highlight the input in red when it does not pass validation.',
			},
		},
	},
	render: (args) => (<SearchWithHooks {...args} />)
};

export const Placeholder: Story = {
	args: {
		label: 'Placeholder',
		name: 'Placeholder',
		id: 'Placeholder',
		placeholder: 'Placeholder',
	},
	parameters: {
		docs: {
			description: {
				story: 'Used to show an example of a value that can be entered into an input',
			},
		},
	},
	render: (args) => (<SearchWithHooks {...args} />)
};

export const EventCallbacks: Story = {
	argTypes: {
		onFocus: {
			action: 'Focused',
		},
		onBlur: {
			action: 'Blured',
		},
		onSubmit: {
			action: 'Submitted',
		},
	},
	args: {
		name: 'event callbacks',
		id: 'event callbacks',
		label: 'submit, focus, blur event',
		value: '',
	},
	parameters: {
		docs: {
			description: {
				story: 'The callback function can be provided to the onFocus, onBlur onSubmit event',
			},
		},
	},
	render: (args) => (<SearchWithHooks {...args} />)
};

export const LabelHidden: Story = {
	args: {
		label: 'LabelHidden',
		name: 'LabelHidden',
		id: 'LabelHidden',
		labelHidden: true,
	},
	parameters: {
		docs: {
			description: {
				story: 'Makes the label visible only to screen readers',
			},
		},
	},
	render: (args) => (<SearchWithHooks {...args} />)
};

export const Error: Story = {
	args: {
		label: 'Error',
		name: 'Error',
		id: 'Error',
		error: 'error messsage',
	},
	parameters: {
		docs: {
			description: {
				story: 'Displays an error message under the input',
			},
		},
	},
	render: (args) => (<SearchWithHooks {...args} />)
};

type Variant = {
	label: Props['label']
	isDisabled?: boolean,
	isInvalid?: boolean,
	isReadOnly?: boolean,
	labelHidden?: boolean,
	error?: Props['error'],
	placeholder?: Props['placeholder'],
}

const variants: Variant[] = [
	{
		label: 'disabled',
		isDisabled: true,
	},
	{
		label: 'invalid',
		isInvalid: true,
	},
	{
		label: 'read only',
		isReadOnly: true,
	},
	{
		label: 'label hidden',
		labelHidden: true,
	},
	{
		label: 'error',
		error: 'error',
	},
	{
		label: 'placeholder',
		placeholder: 'placeholder',
	},
]

export const All: Story = {
	parameters: {
		docs: {
			description: {
				story: 'All possible variants of input states are presented here',
			},
		},
	},
	render: () => (
		<>
			{variants.map((variant) => (
				<SearchWithHooks
					className='mb-6 last:mb-0'
					key={variant.label}
					id={variant.label}
					name={variant.label}
					label={variant.label}
					isDisabled={variant.isDisabled}
					isInvalid={variant.isInvalid}
					isReadOnly={variant.isReadOnly}
					labelHidden={variant.labelHidden}
					error={variant.error}
					onSubmit={() => { }}
					placeholder={variant.placeholder}
				/>
			))}
		</>
	),
};