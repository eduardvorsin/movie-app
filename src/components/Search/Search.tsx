'use client';
import { ChangeEventHandler, FocusEventHandler, FormEventHandler } from 'react';
import Button from '@/components/Button/Button';
import InlineMessage from '@/components/InlineMessage/InlineMessage';
import { GeneralProps } from '@/types/shared';

export type Props = {
	value: string,
	id: string,
	label: string,
	name: string,
	onChange: ChangeEventHandler<HTMLInputElement>,
	onSubmit: FormEventHandler<HTMLFormElement>,
	onFocus?: FocusEventHandler<HTMLInputElement>,
	onBlur?: FocusEventHandler<HTMLInputElement>,
	labelHidden?: boolean,
	isDisabled?: boolean,
	isReadOnly?: boolean,
	isInvalid?: boolean,
	placeholder?: string,
	error?: string,
	dictionary: Record<'button', string>,
} & GeneralProps;

export default function Search({
	id,
	className,
	name,
	testId,
	isDisabled,
	isReadOnly,
	isInvalid,
	placeholder,
	label,
	labelHidden,
	error,
	value,
	onChange,
	onFocus,
	onBlur,
	onSubmit,
	dictionary,
	...props
}: Props) {
	const labelClasses = [
		'block text-100 mb-1 text-neutral-1000 dark:text-dark-neutral-1000 transition-colors duration-150',
		labelHidden ? 'sr-only' : '',
	].join(' ');

	const inputClasses = [
		'outline-none w-full h-[2.25rem] py-2 pl-[0.375rem] pr-8 bg-neutral-0 dark:bg-dark-neutral-200 border-neutral-300 dark:border-dark-neutral-350 text-neutral-1000 dark:text-dark-neutral-900 rounded rounded-[0.1875rem] border-2 text-100 enabled:hover:bg-neutral-100 enabled:hover:border-neutral-400 dark:enabled:hover:bg-dark-neutral-250 dark:enabled:hover:border-dark-neutral-350 focus:border-blue-500 dark:focus:border-blue-300 transition-colors duration-150 ',
		isDisabled ? 'opacity-disabled cursor-not-allowed' : '',
		isInvalid ? 'border-red-600 dark:border-red-300' : 'border-neutral-300 dark:border-dark-neutral-350',
	].join(' ');

	return (
		<div
			className={className}
			data-testid={testId}
			{...props}
		>
			<form
				onSubmit={onSubmit}
			>
				<div
					role='search'
				>
					<label
						className={labelClasses}
						htmlFor={id}
					>
						{label}
					</label>
					<div
						className='relative'
					>
						<input
							className={inputClasses}
							id={id}
							type='text'
							name={name}
							placeholder={placeholder}
							disabled={isDisabled}
							readOnly={isReadOnly}
							onChange={onChange}
							value={value}
							onFocus={onFocus}
							onBlur={onBlur}
							aria-invalid={isInvalid}
						/>
						<Button
							appearance='ghost'
							className='absolute top-1/2 right-[0.375rem] -translate-y-1/2'
							isDisabled={isDisabled}
							iconButton
						>
							<svg
								className='w-6 h-6 p-0.5 fill-current'
								viewBox='0 0 32 32'
							>
								<use href={'/assets/icons/loupe.svg#loupe'}></use>
							</svg>
							{dictionary.button}
						</Button>
					</div>
				</div>
			</form>
			{error && (
				<InlineMessage
					className='mt-1'
					message={error}
					fieldId={`${id}-error-message`}
					appearance='error'
				/>
			)}
		</div>
	);
}