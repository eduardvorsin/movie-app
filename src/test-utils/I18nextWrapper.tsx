import React, { ReactNode } from 'react';
import i18next from '@/i18n/client';
import { I18nextProvider } from 'react-i18next';

type Props = {
	children: ReactNode,
}

const I18nextWrapper = ({
	children,
}: Props) => {
	return (
		<I18nextProvider i18n={i18next}>
			{children}
		</I18nextProvider>
	);
};

export default I18nextWrapper;