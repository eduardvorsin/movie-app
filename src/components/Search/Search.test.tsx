import { render, screen } from '@testing-library/react';
import Search from './Search';
import userEvent from '@testing-library/user-event';
import i18next from "@/i18n/client";
import I18nextWrapper from "@/test-utils/I18nextWrapper";

describe('Search tests', () => {
	it('is rendered correctly', () => {
		const dictionary = { button: 'To find' };
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn((e) => e.preventDefault());
		render(
			<Search
				name='test-search'
				id='test-search'
				label='test-search'
				value=''
				testId='test-search'
				onChange={changeMockFn}
				clearButton={false}
				onSubmit={submitMockFn}
				dictionary={dictionary}
			/>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-search')).toBeInTheDocument();
	});

	it('when submitting the form, the mock function is triggered', async () => {
		const dictionary = { button: 'To find' };
		const user = userEvent.setup();
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn((e) => e.preventDefault());
		render(
			<Search
				name='test-search'
				id='test-search'
				label='test-search'
				value=''
				testId='test-search'
				onChange={changeMockFn}
				clearButton={false}
				onSubmit={submitMockFn}
				dictionary={dictionary}
			/>
		);

		await user.click(screen.getByRole<HTMLButtonElement>('button'));

		expect(submitMockFn).toHaveBeenCalledTimes(1);
	});

	it('when entering text into the input, the mock function is triggered', async () => {
		const dictionary = { button: 'To find' };
		const user = userEvent.setup();
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn((e) => e.preventDefault());
		render(
			<Search
				name='test-search'
				id='test-search'
				label='test-search'
				value=''
				testId='test-search'
				onChange={changeMockFn}
				clearButton={false}
				onSubmit={submitMockFn}
				dictionary={dictionary}
			/>
		);

		await user.type(screen.getByRole<HTMLInputElement>('textbox'), 'a');

		expect(changeMockFn).toHaveBeenCalledTimes(1);
	});

	it('when the input receives the focus, the mock function is triggered', async () => {
		const dictionary = { button: 'To find' };
		const user = userEvent.setup();
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn((e) => e.preventDefault());
		const focusMockFn = jest.fn();
		render(
			<Search
				name='test-search'
				id='test-search'
				label='test-search'
				value=''
				testId='test-search'
				onChange={changeMockFn}
				clearButton={false}
				onSubmit={submitMockFn}
				onFocus={focusMockFn}
				dictionary={dictionary}
			/>
		);

		await user.tab();

		expect(focusMockFn).toHaveBeenCalledTimes(1);
	});

	it('when the input loses focus, the mock function is triggered', async () => {
		const dictionary = { button: 'To find' };
		const user = userEvent.setup();
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn((e) => e.preventDefault());
		const blurMockFn = jest.fn();
		render(
			<Search
				name='test-search'
				id='test-search'
				label='test-search'
				value=''
				testId='test-search'
				onChange={changeMockFn}
				clearButton={false}
				onSubmit={submitMockFn}
				onFocus={blurMockFn}
				dictionary={dictionary}
			/>
		);

		await user.tab();
		await user.tab();

		expect(blurMockFn).toHaveBeenCalledTimes(1);
	});

	it('when isDisabled is true, the input and the button become inactive', async () => {
		const dictionary = { button: 'To find' };
		const user = userEvent.setup();
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn((e) => e.preventDefault());
		render(
			<Search
				name='test-search'
				id='test-search'
				label='test-search'
				value=''
				isDisabled
				testId='test-search'
				onChange={changeMockFn}
				clearButton={false}
				onSubmit={submitMockFn}
				dictionary={dictionary}
			/>
		);

		await user.type(screen.getByRole<HTMLInputElement>('textbox'), 'a');
		await user.click(screen.getByRole<HTMLButtonElement>('button'));

		expect(changeMockFn).toHaveBeenCalledTimes(0);
		expect(submitMockFn).toHaveBeenCalledTimes(0);
	});

	it('with IsReadOnly set to true, the mock function is triggered when the focus is on, but not when the value is changed', async () => {
		const dictionary = { button: 'To find' };
		const user = userEvent.setup();
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn((e) => e.preventDefault());
		const focusMockFn = jest.fn();
		render(
			<Search
				name='test-search'
				id='test-search'
				label='test-search'
				value=''
				isReadOnly
				testId='test-search'
				onChange={changeMockFn}
				clearButton={false}
				onSubmit={submitMockFn}
				onFocus={focusMockFn}
				dictionary={dictionary}
			/>
		);

		await user.type(screen.getByRole<HTMLInputElement>('textbox'), 'a');
		await user.tab();

		expect(changeMockFn).toHaveBeenCalledTimes(0);
		expect(focusMockFn).toHaveBeenCalledTimes(1);
	});

	it('if IsInvalid is true, the input\'s border color turns red', () => {
		const dictionary = { button: 'To find' };
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn((e) => e.preventDefault());
		render(
			<Search
				name='test-search'
				id='test-search'
				label='test-search'
				value=''
				testId='test-search'
				isInvalid
				onChange={changeMockFn}
				clearButton={false}
				onSubmit={submitMockFn}
				dictionary={dictionary}
			/>
		);

		expect(screen.getByRole<HTMLInputElement>('textbox')).toHaveClass('border-red-600');
	});

	it('when labelHidden is set to true, the label should be visible only to screen readers', () => {
		const dictionary = { button: 'To find' };
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn((e) => e.preventDefault());
		render(
			<Search
				name='test-search'
				id='test-search'
				label='test-search'
				value=''
				testId='test-search'
				labelHidden
				onChange={changeMockFn}
				clearButton={false}
				onSubmit={submitMockFn}
				dictionary={dictionary}
			/>
		);

		expect(screen.getByText<HTMLLabelElement>('test-search')).toHaveClass('sr-only');
	});

	it('when the error prop is not empty, an error message is displayed next to the input', () => {
		const dictionary = { button: 'To find' };
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn((e) => e.preventDefault());
		render(
			<Search
				name='test-search'
				id='test-search'
				label='test-search'
				value=''
				testId='test-search'
				error='mock error'
				onChange={changeMockFn}
				clearButton={false}
				onSubmit={submitMockFn}
				dictionary={dictionary}
			/>
		);

		expect(screen.getByText<HTMLParagraphElement>('mock error')).toBeInTheDocument();
	});

	it('when you click on the "Clear" button, the mock function should be called', async () => {
		const dictionary = { button: 'To find', clearButton: 'To clear' };
		const user = userEvent.setup();
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn((e) => e.preventDefault());
		const clearMockFn = jest.fn();

		render(
			<Search
				name='test-search'
				id='test-search'
				label='test-search'
				value=''
				testId='test-search'
				onChange={changeMockFn}
				onClear={clearMockFn}
				clearButton={true}
				onSubmit={submitMockFn}
				dictionary={dictionary}
			/>
		);

		await user.click(screen.getByRole<HTMLButtonElement>('button', { name: /to clear/i }));

		expect(clearMockFn).toHaveBeenCalledTimes(1);
	});

	it('is a basic snapshot', () => {
		const dictionary = { button: 'To find' };
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn((e) => e.preventDefault());
		render(
			<Search
				name='test-search'
				id='test-search'
				label='test-search'
				value=''
				testId='test-search'
				onChange={changeMockFn}
				clearButton={false}
				onSubmit={submitMockFn}
				dictionary={dictionary}
			/>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-search')).toMatchSnapshot();
	});

	it('is a snapshot with labelHidden set to true', () => {
		const dictionary = { button: 'To find' };
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn((e) => e.preventDefault());
		render(
			<Search
				name='test-search'
				id='test-search'
				label='test-search'
				value=''
				testId='test-search'
				labelHidden
				onChange={changeMockFn}
				clearButton={false}
				onSubmit={submitMockFn}
				dictionary={dictionary}
			/>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-search')).toMatchSnapshot();
	});

	it('is a snapshot with isDisabled set to true', () => {
		const dictionary = { button: 'To find' };
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn((e) => e.preventDefault());
		render(
			<Search
				name='test-search'
				id='test-search'
				label='test-search'
				value=''
				testId='test-search'
				labelHidden
				isDisabled
				onChange={changeMockFn}
				clearButton={false}
				onSubmit={submitMockFn}
				dictionary={dictionary}
			/>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-search')).toMatchSnapshot();
	});

	it('is snapshot with IsInvalid equal to true', () => {
		const dictionary = { button: 'To find' };
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn((e) => e.preventDefault());
		render(
			<Search
				name='test-search'
				id='test-search'
				label='test-search'
				value=''
				testId='test-search'
				labelHidden
				isInvalid
				onChange={changeMockFn}
				clearButton={false}
				onSubmit={submitMockFn}
				dictionary={dictionary}
			/>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-search')).toMatchSnapshot();
	});

	it('is a snapshot with IsReadOnly equal to true', () => {
		const dictionary = { button: 'To find' };
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn((e) => e.preventDefault());
		render(
			<Search
				name='test-search'
				id='test-search'
				label='test-search'
				value=''
				testId='test-search'
				labelHidden
				isReadOnly
				onChange={changeMockFn}
				clearButton={false}
				onSubmit={submitMockFn}
				dictionary={dictionary}
			/>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-search')).toMatchSnapshot();
	});

	it('is a snapshot with clearButton equal to true', () => {
		const dictionary = { button: 'To find', clearButton: 'To clear' };
		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn((e) => e.preventDefault());
		render(
			<Search
				name='test-search'
				id='test-search'
				label='test-search'
				value=''
				testId='test-search'
				onChange={changeMockFn}
				clearButton={true}
				onSubmit={submitMockFn}
				dictionary={dictionary}
			/>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-search')).toMatchSnapshot();
	});
});

describe('Search integration tests', () => {
	it('localization into English works correctly', async () => {
		await i18next.changeLanguage('en');
		const translation = new RegExp(i18next.t('search.button'));

		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn((e) => e.preventDefault());
		const dictionary = { button: i18next.t('search.button') };
		render(
			<Search
				name='test-search'
				id='test-search'
				label='test-search'
				value=''
				testId='test-search'
				labelHidden
				isReadOnly
				onChange={changeMockFn}
				clearButton={false}
				onSubmit={submitMockFn}
				dictionary={dictionary}
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(screen.getByText<HTMLSpanElement>(translation)).toBeInTheDocument();
	});

	it('localization into Russian works correctly', async () => {
		await i18next.changeLanguage('ru');
		const translation = new RegExp(i18next.t('search.button'));

		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn((e) => e.preventDefault());
		const dictionary = { button: i18next.t('search.button') };
		render(
			<Search
				name='test-search'
				id='test-search'
				label='test-search'
				value=''
				testId='test-search'
				labelHidden
				isReadOnly
				onChange={changeMockFn}
				clearButton={false}
				onSubmit={submitMockFn}
				dictionary={dictionary}
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(screen.getByText<HTMLSpanElement>(translation)).toBeInTheDocument();
	});

	it('is a snapshot with English localization', async () => {
		await i18next.changeLanguage('en');

		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn((e) => e.preventDefault());
		const dictionary = { button: i18next.t('search.button') };
		render(
			<Search
				name='test-search'
				id='test-search'
				label='test-search'
				value=''
				testId='test-search'
				labelHidden
				isReadOnly
				onChange={changeMockFn}
				clearButton={false}
				onSubmit={submitMockFn}
				dictionary={dictionary}
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(screen.getByTestId<HTMLAnchorElement>('test-search')).toMatchSnapshot();
	});

	it('is a snapshot with Russian localization', async () => {
		await i18next.changeLanguage('ru');

		const changeMockFn = jest.fn();
		const submitMockFn = jest.fn((e) => e.preventDefault());
		const dictionary = { button: i18next.t('search.button') };
		render(
			<Search
				name='test-search'
				id='test-search'
				label='test-search'
				value=''
				testId='test-search'
				labelHidden
				isReadOnly
				onChange={changeMockFn}
				clearButton={false}
				onSubmit={submitMockFn}
				dictionary={dictionary}
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(screen.getByTestId<HTMLAnchorElement>('test-search')).toMatchSnapshot();
	});
});