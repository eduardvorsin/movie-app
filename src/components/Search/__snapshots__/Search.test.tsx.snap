// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search tests is a basic snapshot 1`] = `
<div
  data-testid="test-search"
>
  <form>
    <div
      role="search"
    >
      <label
        class="block text-100 mb-1 text-neutral-1000 dark:text-dark-neutral-1000 "
        for="test-search"
      >
        test-search
      </label>
      <div
        class="relative"
      >
        <input
          class="outline-none w-full h-[2.25rem] py-2 pl-[0.375rem] pr-8 bg-neutral-0 dark:bg-dark-neutral-200 border-neutral-300 dark:border-dark-neutral-350 text-neutral-1000 dark:text-dark-neutral-900 rounded rounded-[0.1875rem] border-2 text-100 enabled:hover:bg-neutral-100 enabled:hover:border-neutral-400 dark:enabled:hover:bg-dark-neutral-250 dark:enabled:hover:border-dark-neutral-350 focus:border-blue-500 dark:focus:border-blue-300 transition-colors duration-150 border-neutral-300 dark:border-dark-neutral-350"
          id="test-search"
          type="text"
          value=""
        />
        <button
          class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-dark-neutral-0 enabled:hover:text-dark-neutral-400 enabled:active:text-dark-neutral-600 dark:text-neutral-0 dark:enabled:hover:text-neutral-400 dark:enabled:active:text-neutral-600    absolute top-1/2 right-[0.375rem] -translate-y-1/2"
        >
          <span
            class="inline-flex"
          >
            <svg
              class="w-6 h-6 p-0.5 fill-current"
              viewBox="0 0 32 32"
            >
              <use
                href="undefined#loupe"
              />
            </svg>
            Search
          </span>
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`Search tests is a snapshot with IsReadOnly equal to true 1`] = `
<div
  data-testid="test-search"
>
  <form>
    <div
      role="search"
    >
      <label
        class="block text-100 mb-1 text-neutral-1000 dark:text-dark-neutral-1000 sr-only"
        for="test-search"
      >
        test-search
      </label>
      <div
        class="relative"
      >
        <input
          class="outline-none w-full h-[2.25rem] py-2 pl-[0.375rem] pr-8 bg-neutral-0 dark:bg-dark-neutral-200 border-neutral-300 dark:border-dark-neutral-350 text-neutral-1000 dark:text-dark-neutral-900 rounded rounded-[0.1875rem] border-2 text-100 enabled:hover:bg-neutral-100 enabled:hover:border-neutral-400 dark:enabled:hover:bg-dark-neutral-250 dark:enabled:hover:border-dark-neutral-350 focus:border-blue-500 dark:focus:border-blue-300 transition-colors duration-150 border-neutral-300 dark:border-dark-neutral-350"
          id="test-search"
          readonly=""
          type="text"
          value=""
        />
        <button
          class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-dark-neutral-0 enabled:hover:text-dark-neutral-400 enabled:active:text-dark-neutral-600 dark:text-neutral-0 dark:enabled:hover:text-neutral-400 dark:enabled:active:text-neutral-600    absolute top-1/2 right-[0.375rem] -translate-y-1/2"
        >
          <span
            class="inline-flex"
          >
            <svg
              class="w-6 h-6 p-0.5 fill-current"
              viewBox="0 0 32 32"
            >
              <use
                href="undefined#loupe"
              />
            </svg>
            Search
          </span>
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`Search tests is a snapshot with isDisabled set to true 1`] = `
<div
  data-testid="test-search"
>
  <form>
    <div
      role="search"
    >
      <label
        class="block text-100 mb-1 text-neutral-1000 dark:text-dark-neutral-1000 sr-only"
        for="test-search"
      >
        test-search
      </label>
      <div
        class="relative"
      >
        <input
          class="outline-none w-full h-[2.25rem] py-2 pl-[0.375rem] pr-8 bg-neutral-0 dark:bg-dark-neutral-200 border-neutral-300 dark:border-dark-neutral-350 text-neutral-1000 dark:text-dark-neutral-900 rounded rounded-[0.1875rem] border-2 text-100 enabled:hover:bg-neutral-100 enabled:hover:border-neutral-400 dark:enabled:hover:bg-dark-neutral-250 dark:enabled:hover:border-dark-neutral-350 focus:border-blue-500 dark:focus:border-blue-300 transition-colors duration-150 border-neutral-300 dark:border-dark-neutral-350"
          disabled=""
          id="test-search"
          type="text"
          value=""
        />
        <button
          class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0 opacity-disabled cursor-not-allowed text-dark-neutral-0 enabled:hover:text-dark-neutral-400 enabled:active:text-dark-neutral-600 dark:text-neutral-0 dark:enabled:hover:text-neutral-400 dark:enabled:active:text-neutral-600    absolute top-1/2 right-[0.375rem] -translate-y-1/2"
          disabled=""
        >
          <span
            class="inline-flex"
          >
            <svg
              class="w-6 h-6 p-0.5 fill-current"
              viewBox="0 0 32 32"
            >
              <use
                href="undefined#loupe"
              />
            </svg>
            Search
          </span>
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`Search tests is a snapshot with labelHidden set to true 1`] = `
<div
  data-testid="test-search"
>
  <form>
    <div
      role="search"
    >
      <label
        class="block text-100 mb-1 text-neutral-1000 dark:text-dark-neutral-1000 sr-only"
        for="test-search"
      >
        test-search
      </label>
      <div
        class="relative"
      >
        <input
          class="outline-none w-full h-[2.25rem] py-2 pl-[0.375rem] pr-8 bg-neutral-0 dark:bg-dark-neutral-200 border-neutral-300 dark:border-dark-neutral-350 text-neutral-1000 dark:text-dark-neutral-900 rounded rounded-[0.1875rem] border-2 text-100 enabled:hover:bg-neutral-100 enabled:hover:border-neutral-400 dark:enabled:hover:bg-dark-neutral-250 dark:enabled:hover:border-dark-neutral-350 focus:border-blue-500 dark:focus:border-blue-300 transition-colors duration-150 border-neutral-300 dark:border-dark-neutral-350"
          id="test-search"
          type="text"
          value=""
        />
        <button
          class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-dark-neutral-0 enabled:hover:text-dark-neutral-400 enabled:active:text-dark-neutral-600 dark:text-neutral-0 dark:enabled:hover:text-neutral-400 dark:enabled:active:text-neutral-600    absolute top-1/2 right-[0.375rem] -translate-y-1/2"
        >
          <span
            class="inline-flex"
          >
            <svg
              class="w-6 h-6 p-0.5 fill-current"
              viewBox="0 0 32 32"
            >
              <use
                href="undefined#loupe"
              />
            </svg>
            Search
          </span>
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`Search tests is snapshot with IsInvalid equal to true 1`] = `
<div
  data-testid="test-search"
>
  <form>
    <div
      role="search"
    >
      <label
        class="block text-100 mb-1 text-neutral-1000 dark:text-dark-neutral-1000 sr-only"
        for="test-search"
      >
        test-search
      </label>
      <div
        class="relative"
      >
        <input
          aria-invalid="true"
          class="outline-none w-full h-[2.25rem] py-2 pl-[0.375rem] pr-8 bg-neutral-0 dark:bg-dark-neutral-200 border-neutral-300 dark:border-dark-neutral-350 text-neutral-1000 dark:text-dark-neutral-900 rounded rounded-[0.1875rem] border-2 text-100 enabled:hover:bg-neutral-100 enabled:hover:border-neutral-400 dark:enabled:hover:bg-dark-neutral-250 dark:enabled:hover:border-dark-neutral-350 focus:border-blue-500 dark:focus:border-blue-300 transition-colors duration-150 border-red-600 dark:border-red-300"
          id="test-search"
          type="text"
          value=""
        />
        <button
          class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-0  text-dark-neutral-0 enabled:hover:text-dark-neutral-400 enabled:active:text-dark-neutral-600 dark:text-neutral-0 dark:enabled:hover:text-neutral-400 dark:enabled:active:text-neutral-600    absolute top-1/2 right-[0.375rem] -translate-y-1/2"
        >
          <span
            class="inline-flex"
          >
            <svg
              class="w-6 h-6 p-0.5 fill-current"
              viewBox="0 0 32 32"
            >
              <use
                href="undefined#loupe"
              />
            </svg>
            Search
          </span>
        </button>
      </div>
    </div>
  </form>
</div>
`;
